### PR TITLE
fix: don't remove the prefix from the request path

### PR DESCRIPTION
**Pull request by jinwoo**
The request path is used later when pushing related resources and push
doesn't work if a path is not absolute.

Also don't remove the query part of the url either because related
resources may be different depending on the query part.

Also importantly, this PR fixes the cookie path for the cache cookie. It
must be set to the root path (/). If `path` is not specified, it
defaults to the current request path, which is wrong.

Additional changes:
- revert to `require('../..')` in the sample app. `npm install` for file
  URLs is brittle and often fails with cryptic errors.
- update packages.

### CODE REVIEW

**Comment by jinwoo** [comment]
I also fixed the wrong cookie path set for the cache cookie.

**Comment by jinwoo** [comment]
@mcollina Would you mind taking a look at this PR?

File: samples/static-page/index.js [hunk 1]
```diff
@@ -14,7 +14,8 @@
 
 const {ArgumentParser} = require('argparse');
 const fastify = require('fastify');
-const fastifyAutoPush = require('fastify-auto-push');
+// This should be `require('fastify-auto-push')` for stand-alone apps.
```
Comment (JustinBeckwith) [inline samples/static-page/index.js]: To get around this, I like to just reference the module name, and then tell people to run `npm link && npm link foo` when they run the samples locally.  Makes it easier for folks who will invariably just copy and paste.

File: ts/src/index.ts [hunk 2]
```diff
@@ -57,25 +57,23 @@ async function staticServeFn(
...
 
   app.register(fastifyStatic, opts);
 
   app.addHook('onRequest', async (req, res) => {
     if (isHttp2Request(req)) {
       const reqStream = req.stream;
-      const url: string = req.url;
-      let reqPath: string = url.split('?')[0];
-      reqPath = reqPath.replace(prefix, '');
+      const reqPath: string = req.url;
```
Comment (JustinBeckwith) [inline ts/src/index.ts]: Why do we need to explicitly call out the type here?

File: samples/static-page/index.js (hunk 1 above)
Comment (jinwoo) [reply samples/static-page/index.js]: SGTM. Reverted.

File: ts/src/index.ts (hunk 2 above)
Comment (jinwoo) [reply ts/src/index.ts]: We don't. Deleted. Thanks!

**Comment by jinwoo** [comment]
Merging this now. @mcollina, please let me know if you have comments on this PR. I'll address them with a follow-up PR.

**Comment by mcollina** [comment]
Sorry, I’ve been traveling from the last week to the end of this week :/.
I’ll have a look after I came back.
Il giorno mar 20 feb 2018 alle 18:40 Jinwoo Lee <notifications@github.com>
ha scritto:

**Pull request merged**